### DISCUSSION TITLE
nextjs 13 react query ssr problem in app directory

### DISCUSSION THREAD

**Discussion by airHalfSoundHalf** [community/community#62079] [Programming Help]
### Select Topic Area

Question

### Feature Area

Issues

### Body

I am using next js 13 app directory in my project.
If I put the query client as a parameter in the dehydrate, the following error appears. Does anyone know about this bug? help me if you have.

```terminal
types have separate declarations of a private property 'querycache'
```

```tsx
// app/getQueryClient.jsx

import { QueryClient } from '@tanstack/react-query'
import { cache } from 'react'

const getQueryClient = cache(() => new QueryClient())
export default getQueryClient
```

```tsx
// app/posts/hydratedPosts.jsx

export default async function HydratedPosts() {
  const queryClient = getQueryClient()
  await queryClient.prefetchQuery(['posts'], getPosts)
ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡhereㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ
  const dehydratedState = dehydrate(queryClient)
                                        ⬆️

  return (
    <Hydrate state={dehydratedState}>
      <Posts />
    </Hydrate>
  )
}
```

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @airHalfSoundHalf ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
